src/main/java/org/asciidoctor/maven/site/AsciidoctorParser.java

```diff
@@ -31,31 +38,134 @@
+    }
+
+    protected OptionsBuilder initOptions(MavenProject project, File siteDirectory) {
+        return OptionsBuilder.options()
+                .backend("xhtml")
+                .safe(SafeMode.UNSAFE)
+                .baseDir(new File(siteDirectory, ROLE_HINT));
+    }
+
+    protected AttributesBuilder initAttributes(MavenProject project, File siteDirectory) {
+        return AttributesBuilder.attributes()
+            .attribute("idprefix", "@")
+            .attribute("showtitle", "@");
+    }
+
+    protected OptionsBuilder processAsciiDocConfig(Xpp3Dom siteConfig, OptionsBuilder options, AttributesBuilder attributes) {
+        if (siteConfig == null) {
+            return options.attributes(attributes);
+        }
+
+        Xpp3Dom asciidocConfig = siteConfig.getChild("asciidoc");
+        if (asciidocConfig == null) {
+            return options.attributes(attributes);
+        }
+
+        for (Xpp3Dom asciidocOpt : asciidocConfig.getChildren()) {
+            String optName = asciidocOpt.getName();
+            if ("attributes".equals(optName)) {
+                for (Xpp3Dom asciidocAttr : asciidocOpt.getChildren()) {
+                    attributes.attribute(asciidocAttr.getName(), asciidocAttr.getValue());
+                }
+            }
+            else if ("requires".equals(optName)) {
+                Xpp3Dom[] requires = asciidocOpt.getChildren("require");
+                // supports variant:
+                // <requires>
+                //     <require>time</require>
+                // </requires>
+                if (requires.length > 0) {
+                    for (Xpp3Dom require : requires) {
+                        requireLibrary(require.getValue());
+                    }
+                }
+                else {
+                    // supports variant:
+                    // <requires>time,base64</requires>
+                    for (String require : asciidocOpt.getValue().split(",")) {
+                        requireLibrary(require);
+                    }
+                }
+            }
+            // TODO support multiple template directories
+            else if ("templateDir".equals(optName) || "template_dir".equals(optName)) {
+                options.templateDir(new File(project.getBasedir(), asciidocOpt.getValue()));
+            }
+            else {
+                options.option(optName.replaceAll("(?<!_)([A-Z])", "_$1").toLowerCase(), asciidocOpt.getValue());
+            }
+        }
+        return options.attributes(attributes);
+    }
+
+    protected String convertAsciiDoc(String source, OptionsBuilder options) {
+        return asciidoctor.convert(source, options);
+    }
+
+    private void requireLibrary(String require) {
+        if (!require.trim().isEmpty()) {
+            try {
+                // FIXME AsciidoctorJ should provide a public API for requiring paths in the Ruby runtime
+                RubyUtils.requireLibrary(JRubyRuntimeContext.get(), require.trim());
```

Comment:
I'll change this to a TODO since we haven't upgraded the AsciidoctorJ version yet. As soon as we do, we'll switch it over.